assert isinstance(request, HttpRequest) prevents use with Django Rest Framework (drf)

The `assert` in https://github.com/cnobile2012/django-pam/blob/0d854b90d02022a5d57dba984ea551696ef95228/django_pam/auth/backends.py#L52 fails when used in a DRF app when accessing the `/api-token-auth` page (see https://www.django-rest-framework.org/api-guide/authentication/).

`request` is then  a `rest_framework.request.Request`.
If the assert is commented out, it works as expected.

IMHO the assert not required in the code, as `request` isn't used in that method at all. So a condition on it to be from a fixed type is not necessary.

Hello, I'm really sorry I haven't replied. Please reopen. I have a stack trace:
```
ERROR Internal Server Error: /api-token-auth/
Traceback (most recent call last):
  File "/usr/local/lib/python3.6/site-packages/django/core/handlers/exception.py", line 34, in inner
    response = get_response(request)
  File "/usr/local/lib/python3.6/site-packages/django/core/handlers/base.py", line 126, in _get_response
    response = self.process_exception_by_middleware(e, request)
  File "/usr/local/lib/python3.6/site-packages/django/core/handlers/base.py", line 124, in _get_response
    response = wrapped_callback(request, *callback_args, **callback_kwargs)
  File "/usr/local/lib/python3.6/site-packages/django/views/decorators/csrf.py", line 54, in wrapped_view
    return view_func(*args, **kwargs)
  File "/usr/local/lib/python3.6/site-packages/django/views/generic/base.py", line 68, in view
    return self.dispatch(request, *args, **kwargs)
  File "/usr/local/lib/python3.6/site-packages/rest_framework/views.py", line 495, in dispatch
    response = self.handle_exception(exc)
  File "/usr/local/lib/python3.6/site-packages/rest_framework/views.py", line 455, in handle_exception
    self.raise_uncaught_exception(exc)
  File "/usr/local/lib/python3.6/site-packages/rest_framework/views.py", line 492, in dispatch
    response = handler(request, *args, **kwargs)
  File "/usr/local/lib/python3.6/site-packages/rest_framework/authtoken/views.py", line 44, in post
    serializer.is_valid(raise_exception=True)
  File "/usr/local/lib/python3.6/site-packages/rest_framework/serializers.py", line 236, in is_valid
    self._validated_data = self.run_validation(self.initial_data)
  File "/usr/local/lib/python3.6/site-packages/rest_framework/serializers.py", line 437, in run_validation
    value = self.validate(value)
  File "/usr/local/lib/python3.6/site-packages/rest_framework/authtoken/serializers.py", line 21, in validate
    username=username, password=password)
  File "/usr/local/lib/python3.6/site-packages/django/contrib/auth/__init__.py", line 73, in authenticate
    user = backend.authenticate(request, **credentials)
  File "/usr/local/lib/python3.6/site-packages/django_pam/auth/backends.py", line 52, in authenticate
    "The 'request' positonal argument should be either None or an "
AssertionError: The 'request' positonal argument should be either None or an HttpRequest object.
[28/Mar/2019 17:56:07] "POST /api-token-auth/ HTTP/1.1" 500 15677
```
Python package versions:
* Django==2.1.5
* djangorestframework==3.9.1
* django-pam==1.5.0

My point is, that the `request` object in the `authenticate()` method isn't used at all. So there is no need to put such a strict requirement (being of a certain type) on it.
The `request` argument _is_ filled, but instead of a subclass of `django.http.HttpRequest` a `rest_framework.request.Request` is used. There is no problem with that, following duck typing.

--- a/django_pam/auth/backends.py
+++ b/django_pam/auth/backends.py
@@ -48,9 +48,6 @@ class PAMBackend(ModelBackend):
         :type extra_fields: dict
         :rtype: The Django user object or `None` if it fails.
         """
-        assert request is None or isinstance(request, HttpRequest), (
-            "The 'request' positonal argument should be either None or an "
-            "HttpRequest object.")
         UserModel = get_user_model()
         user = None
         service = extra_fields.pop('service', 'login')

